submissions/nadsatt/friends-app/modules/filter-items.js

```diff
@@ -0,0 +1,173 @@
+import svgs from './svgs.js';
+
+class Filter {
+    constructor(optionsHTML, category, property, userService){
+        this.optionsHTML = optionsHTML;
+        this.category = category;
+        this.property = property;
+        this.name = `${property}-${category}`;
+        this.userService = userService;
+
+        this.defineElement();
+        this.linkElementWithInstance();
+
+        this.activatedState = 'none';
+        this.activatedOptionElement = this.element.querySelector(`[data-state="none"]`);
+    }
+
+    defineElement(){
+        const innerHTML =
+            `<div class="filter-item__options">
+                ${this.optionsHTML}
+                <span class="filter-item__option" data-state="none">${svgs.none}</span>
+             </div>`;
+
+        this.element = document.createElement('li');
+        this.element.innerHTML = innerHTML;
+
+        this.element.classList.add('filter-item', `${this.category}-filter-item`);
+        this.element.setAttribute('data-current-state', 'none');
+        this.element.setAttribute('data-filter-name', this.name);
+    }
+
+    linkElementWithInstance(){
+        this.element.instance = this;
```

Comment:
I thought about using shadow dom, so my classes will represent corresponding html elements and there will be no need in linking class instance to element and vice versa.

Reply:
Yeap. Looks fine.

But this will not solve places where you traversing thru tree to find DOM element for it's instance.
How you can change such things?

Reply:
But in that case each dom element will be an instance of corresponding class and it removes need in linking instance with dom element/dom element with instance. If I misunderstood your question, please rephrase it or provide me link on code you're talking about. Thanks 😊

Reply:
Ah, I see. I've missed this part. Yeah, let's try what you suggested - we will see where it goes :)